clayui.com/src/components/Editor/index.js

```diff
@@ -54,7 +54,40 @@ const Editor = ({
 		];
 	}
 
-	const [collapseCode, setCollapseCode] = useState(false);
+	function useStickyState(defaultValue, key) {
+		const inBrowser = typeof window !== 'undefined';
+
+		const localStorage = {
+			getItem(key) {
+				return inBrowser ? window.localStorage.getItem(key) : null;
+			},
+
+			setItem(key, value) {
+				if (inBrowser) {
+					window.localStorage.setItem(key, value);
+				}
+			},
+		};
+
+		const [value, setValue] = React.useState(() => {
+			const stickyValue = localStorage.getItem(key);
+
+			return stickyValue !== null
+				? JSON.parse(stickyValue)
+				: defaultValue;
+		});
+
+		React.useEffect(() => {
+			localStorage.setItem(key, JSON.stringify(value));
+		}, [key, value]);
```

Comment:
Whats the purpose of the `JSON.parse` and `JSON.stringify` here? Since our value is just a boolean value, I think we would be fine to not use parse and stringify.

Reply:
Given that local storage are [always strings](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage), using `JSON` ensures you get the same type back out that you put in.

If you pass in boolean values `true`/`false`, they get stored as (effectively) `'true'`/`'false'`, which are both truthy when you read them back out, so the other places in the code where we're doing things like `!collapsed` won't work as expected.

Reply:
@wincent sounds good!